python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/daemon_tests/test_e2e.py

```diff
@@ -547,3 +583,24 @@ def test_custom_condition() -> None:
             _execute_ticks(context, executor)
             runs = _get_runs_for_latest_ticks(context)
             assert len(runs) == 0
+
+
+def test_massive_user_code(capsys) -> None:
+    with get_grpc_workspace_request_context(
+        "massive_user_code"
+    ) as context, get_threadpool_executor() as executor:
+        freeze_dt = datetime.datetime(2024, 8, 16, 1, 35)
+
+        for _ in range(2):
+            clock_time = time.time()
+            with freeze_time(freeze_dt):
+                _execute_ticks(context, executor)
+                runs = _get_runs_for_latest_ticks(context)
+                assert len(runs) == 0
+            duration = time.time() - clock_time
+            assert duration < 40.0
+
+            freeze_dt += datetime.timedelta(minutes=1)
+
+    for line in capsys.readouterr():
+        assert "RESOURCE_EXHAUSTED" not in line
```

Comment:
maybe some comments connecting to [1] - or at least what this test assumes about the state of that thing ie "700 defs is ok" . Maybe encode size in name instead of "massive" 


should we have a checked in expect exception test for exceeding the limit also ?

Reply:
The next PR in the stack (forgot to add a reviewer to that one, my bad) adds that explicit exception, and I updated the name to match the value that it will have in that PR.

Perhaps a silly way of arranging my PRs, just wanted to demonstrate (at least to myself) that 500 is not a super-tight bound here, hence 700 in this PR and 500 in the next